### PR TITLE
less noisy log for azure

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

For https://github.com/istio/istio/discussions/52369

1. avoid unmarshall issue when `169.254.169.254` is not accessable.
2. change log level

### CODE REVIEW

**Comment by zirain** [comment]
cc @keithmattix

File: pkg/bootstrap/platform/azure.go [hunk 1]
```diff
@@ -104,30 +104,32 @@ func (e *azureEnv) parseMetadata(metadata string) {
...
 
+const emptyJSONResponse = "{}"
+
 // Generic Azure metadata GET request helper for the response body
 // Uses the default timeout for the HTTP get request
 func metadataRequest(query string) string {
 	client := http.Client{Timeout: defaultTimeout}
 	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s?%s", AzureInstanceURL, query), nil)
 	if err != nil {
-		log.Warnf("Failed to create HTTP request: %v", err)
```
Comment (keithmattix) [inline pkg/bootstrap/platform/azure.go:113]: Wouldn't a lack of access to the metadata service break locality loadbalancing and some telemetry? Warn feels like the correct log level

Comment (zirain) [reply pkg/bootstrap/platform/azure.go:113]: I thinks we did same thing to AWS and GCP?

Comment (keithmattix) [reply pkg/bootstrap/platform/azure.go:113]: Oh if so then disregard. I'm all for consistency

**Review by keithmattix — APPROVED** [review]

**Pull request merged**